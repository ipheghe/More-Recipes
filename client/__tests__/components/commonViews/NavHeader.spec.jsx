import expect from 'expect';
import React from 'react';
import { mount, shallow } from 'enzyme';
import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedNavHeader, { PureNavHeader }
  from '../../../src/commonViews/NavHeader';
import mockAuthCheck from '../../__mocks__/mockAuthCheck';


const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const initialState = {
  auth: {
    error: 'Invalid username',
    message: '',
    userData: {},
    authenticated: true,
    categories: []
  }
};

const store = mockStore(initialState);

const props = {
  authenticated: true
};

/**
 * @param { boolean } isAuthenticated
 *
 * @return { * } null
 */
const setup = (isAuthenticated) => {
  props.authenticated = isAuthenticated;
  const mountedWrapper = mount(<Provider store={store} >
    <ConnectedNavHeader {...props} />
  </Provider>);
  const shallowWrapper = shallow(<PureNavHeader {...props} />);
  return {
    mountedWrapper,
    shallowWrapper
  };
};

describe('<NavHeader', () => {
  beforeEach(() => {
    mockAuthCheck();
  });

  it(`renders UserNavHeader component without 
      crashing if user is authenticated`, () => {
      const { mountedWrapper } = setup(true);
      expect(mountedWrapper).toBeDefined();
      expect(mountedWrapper.find('UserNavHeader').length).toBe(1);
      expect(mountedWrapper.find('MainHeader').length).toBe(0);
      expect(mountedWrapper.exists()).toBe(true);
    });

  it('renders MainHeader component without' +
  'crashing if user is not authenticated ', () => {
    const { shallowWrapper } = setup(false);
    expect(shallowWrapper).toBeDefined();
    expect(shallowWrapper.find('UserNavHeader').length).toBe(0);
    expect(shallowWrapper.find('MainHeader').length).toBe(1);
    expect(shallowWrapper.exists()).toBe(true);
  });

  it('should match component snapshot', () => {
    const tree = render.create(<Router ><PureNavHeader {...props} /></Router>);
    expect(tree).toMatchSnapshot();
  });
});
