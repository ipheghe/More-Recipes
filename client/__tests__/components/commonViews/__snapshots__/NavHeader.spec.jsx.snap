// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavHeader should match component snapshot 1`] = `
<div>
  <div>
    <header>
      <nav
        className="navbar navbar-toggleable-md fixed-top navbar-light bg-faded"
        id="brand-color"
      >
        <button
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler navbar-toggler-right"
          data-target="#navbarSupportedContent"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <a
          className="navbar-brand"
          href="/"
        >
          <img
            alt="logo"
            className="d-inline-block align-top"
            height="30"
            src="test-file-stub"
            width="30"
          />
          MoreRecipes
        </a>
        <div
          className="collapse navbar-collapse"
          id="navbarSupportedContent"
        >
          <ul
            className="navbar-nav mr-auto"
          >
            <li
              className="nav-item nav-home"
              onClick={[Function]}
            >
              <a
                aria-current="true"
                className="nav-link active"
                href="#/"
                onClick={[Function]}
                style={Object {}}
              >
                Home
              </a>
              <span
                className="sr-only"
              >
                (current)
              </span>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={false}
                className="nav-link"
                href="#/login"
                onClick={[Function]}
                style={undefined}
              >
                Login
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={false}
                className="nav-link"
                href="#/signup"
                onClick={[Function]}
                style={undefined}
              >
                SignUp
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </header>
  </div>
</div>
`;
